### PR TITLE
feat: add the GetNfscCard component to components/Templates

### PR DESCRIPTION
**Pull request by luistv-swe**
Add the GetNfscCard component to components/Templates and a story for it

### CODE REVIEW

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }>;
+
+let cardTitle = "Get $NFSC";
+
+export function GetNfscCard(
+  { onClickChip, value, children, className, ...props }: GetNfscCardProps) {
+    let chips = [20, 50, 100].map(
+      (chipValue) => (<Chip selected={value === chipValue} onSelect={() => onClickChip(chipValue)} >{chipValue}</Chip>));
+  return (
+    <div className={cn('flex flex-col justify-start items-center rounded-3.75 gap-3 py-9 px-11.5 bg-black-1000 w-[613px] h-[581px]', className)} {...props}>
```
Comment (samishal1998) [inline src/components/Templates/GetNfscCard/GetNfscCard.tsx:18]: we shouldn't use strict sizing on parts of the layout, strict sizing could be used on small components or icons.
in most cases you might need strict sizing in height only, but depending on the case it's best to use max and min width.
so for this example, I think the design intention was `max-w-[613px] min-h-[581px]`

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    onClickChip: any,
+    value: number
+  }>;
+
+let cardTitle = "Get $NFSC";
+
+export function GetNfscCard(
+  { onClickChip, value, children, className, ...props }: GetNfscCardProps) {
+    let chips = [20, 50, 100].map(
+      (chipValue) => (<Chip selected={value === chipValue} onSelect={() => onClickChip(chipValue)} >{chipValue}</Chip>));
```
Comment (samishal1998) [inline src/components/Templates/GetNfscCard/GetNfscCard.tsx:16]: when you map, it's always best to use key on the elements

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  Partial<{
+    onClickChip: any,
+    value: number
+  }>;
+
+let cardTitle = "Get $NFSC";
+
+export function GetNfscCard(
+  { onClickChip, value, children, className, ...props }: GetNfscCardProps) {
+    let chips = [20, 50, 100].map(
```
Comment (samishal1998) [inline src/components/Templates/GetNfscCard/GetNfscCard.tsx:15]: this is bad for performance, In react the component function is called multiple times, so in this case on each call a new array `[20, 50, 100]` is constructed and in turn the 3 three chip components are re constructed.
in these kind of cases you have 3 solutions:
- A memoized `ChipGroup` component that takes an array of the numbers.
- useMemo on the array.
- (best for this case) the chips numbers should be a prop and memoize GetNfscCard.
- (easiest fix for this case) define a top-level variable outside the component for the numbers array

**Review by samishal1998 — CHANGES_REQUESTED** [review]

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx (hunk 2 above)
Comment (luistv-swe) [reply src/components/Templates/GetNfscCard/GetNfscCard.tsx:16]: Now using chipValue as the key.

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx (hunk 3 above)
Comment (luistv-swe) [reply src/components/Templates/GetNfscCard/GetNfscCard.tsx:15]: Now using chipValues as a prop to be memoized by app

File: src/components/Templates/GetNfscCard/GetNfscCard.tsx (hunk 1 above)
Comment (luistv-swe) [reply src/components/Templates/GetNfscCard/GetNfscCard.tsx:18]: Now using max-w and min-h instead of w- and h-

**Review by samishal1998 — APPROVED** [review]

**Pull request merged**